### PR TITLE
[CP]: Tower of Hanoi

### LINKED ISSUE #6301: [CP]: Tower of Hanoi
### Description

I would like to provide documentation on the Tower of Hanoi algorithm- stack approach.

### Domain

Competitive Programming

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by areenoverclouds**
<hr>

## Description 📜

This is the documentation explaining the recursive stack approach on the Tower of Hanoi problem.
Fixes #6301

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [x] Competitive Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by areenoverclouds** [comment]
Please review this PR, I closed the previous one because I was unable to make changes to it. I'm sorry and I won't mess it up next time.

File: Competitive_Programming/C++/Stacks & Queues/TowerOfHanoi.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+## Complexity Analysis
+
+* Time Complexity: O(2^n)
+* Space Complexity: O(n)
+
+## Further Information
+
+You can read the mathematics and history behind Tower of Hanoi [here](https://www.hackerearth.com/blog/developers/tower-hanoi-recursion-game-algorithm-explained/).
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Stacks & Queues/TowerOfHanoi.md:98]: References are not allowed, You can remove it.

**Review by ravireddy07 — CHANGES_REQUESTED** [review]
Great work @areenoverclouds , Please make the requested changes

**Review by ravireddy07 — APPROVED** [review]

**Comment by areenoverclouds** [comment]
> /assign

why are you spam commenting assign here? this is a pull request and as far as I know, issues have to be assigned and not PRs?

**Pull request merged**